src/workerd/api/web-socket.c++

```diff
@@ -699,7 +699,7 @@ void WebSocket::ensurePumping(jsg::Lock& js) {
           KJ_FAIL_ASSERT("Unexpected native web socket state", native.state);
         }
       }
-    }, [this](jsg::Lock& js, jsg::Value&& exception) mutable {
+    }, [this, thisHandle = JSG_THIS](jsg::Lock& js, jsg::Value&& exception) mutable {
```

Comment:
As an alternative here, the `JSG_THIS` could be attached to the kj::Promise returned by the `then(...)` rather than captured by the lambda. Would save having to grab it twice. Obviously not a big deal tho.

Reply:
I suppose, longer term, we'd want these to maybe be `JSG_VISITABLE_LAMBDA()`s, so that the garbage collector had visibility into them?  But that seems to be a preexisting issue.